lib/puppet/provider/selinux_port/semanage.rb

```diff
@@ -5,7 +5,37 @@
   # SELinux must be enabled. Is there a way to get a better error message?
   confine selinux: true
 
-  python_command = Facter.value(:selinux_python_command)
+  # Determine the appropriate python command
+  def self.python_command
+    @python_command ||= nil
+    return @python_command unless @python_command.nil?
+
+    # Find the correct version of python on the system
+    python_paths = [
+      '/usr/libexec/platform-python',
+      'python',
+      'python3',
+      'python2'
+    ]
+
+    python_command = nil
+    python_paths.each do |pypath|
+      candidate = Puppet::Util.which(pypath)
+
+      next unless candidate
+      if Puppet::Util::Execution.execute("#{candidate} -c 'import semanage'", failonfail: false).exitstatus.zero?
+        python_command = candidate
+        break
+      end
+    end
```

Comment:
Would this work?

```suggestion
    python_command = python_paths.detect do |pypath|
      Puppet::Util::Execution.execute("#{pypath} -c 'import semanage'", failonfail: false).exitstatus.zero?
    end
```

Reply:
@ekohl I don't think it will.

`failonfail` doesn't do what you think if the command itself doesn't exist so we need the `next unless candidate` bit.